staging/src/k8s.io/apiserver/pkg/authorization/cel/compile.go

```diff
@@ -203,6 +259,42 @@ func convertObjectToUnstructured(obj *authorizationv1.SubjectAccessReviewSpec) m
 			"subresource": obj.ResourceAttributes.Subresource,
 			"name":        obj.ResourceAttributes.Name,
 		}
+
+		if includeFieldSelector && obj.ResourceAttributes.FieldSelector != nil {
+			if len(obj.ResourceAttributes.FieldSelector.Requirements) > 0 {
+				requirements := make([]map[string]interface{}, 0, len(obj.ResourceAttributes.FieldSelector.Requirements))
```

Comment:
@jpbetz is there a better way to lazily adapt an existing instance to cel-expression-accessible ref.Vals? I really dislike reconstructing a deep struct like this

Reply:
Not yet :( CEL's direct support is only "primitive types in addition to lists, maps..." ([xref](https://github.com/google/cel-go/?tab=readme-ov-file#environment-setup)).  We added custom support for unstructured in k8s, and have done everything through that so far.  We have experience in SMD [wrapping go structs](https://github.com/kubernetes-sigs/structured-merge-diff/blob/master/value/valuereflect.go), which we could re-apply here.  I've been wanting to get that in place..